### PR TITLE
Add unit validation functionalities to validate() function

### PR DESCRIPTION
**Pull request by sebastianzwickl**
This PR contains the missing functionalities mentioned in #82 regarding the validation of units. In order to ensure an overview during the review process, I made no extension to the units (related to issue #83 and #76). Only a selected unit was added.

### CODE REVIEW

**Review by sandrinecharousset — APPROVED** [review]

File: nomenclature/definitions/variable/economy/economy.yaml [hunk 1]
```diff
@@ -184,7 +184,7 @@ Price|Final Energy|Residential|Electricity:
                 Prices should include the effect of carbon prices
                 Mean price should reflect the variability of different prices that are accessible
                 to end-users (including regulated prices, prices proposed by different competiting retailers...)
-   unit: US$2010/GJ
+   unit: [euro/kWh, US$2010/GJ]
```
Comment (erikfilias) [inline nomenclature/definitions/variable/economy/economy.yaml:187]: euro without a capital letter?

File: nomenclature/__init__.py [hunk 2]
```diff
@@ -265,3 +282,14 @@ def _validate_directional(x):
...
     return len(x) == 2 and all([i in regions for i in x])
+
+
+def _validate_unit(x):
+    # sub function to filter out variables with valid name
+    for i in reversed(x):  # iterate list reversely due to 'remove' method
+        if i in variables.keys():
+            logger.warning(
+                'Unit for variable %s is not given in %s.', i, variables[
+                    i]['unit'])
```
Comment (danielhuppmann) [inline nomenclature/__init__.py:293]: This is deprecated way to format messages, f-strings are easier to read

```suggestion
                f'Unit for variable {i} is not given in {variables[i]['unit']}.'
```

File: nomenclature/definitions/variable/economy/economy.yaml (hunk 1 above)
Comment (danielhuppmann) [reply nomenclature/definitions/variable/economy/economy.yaml:187]: good point, also raised by @sandrinecharousset at #85 - but this should simply be harmonized in a next iteration, this does not need to hold back merging this PR

Comment (sebastianzwickl) [reply nomenclature/definitions/variable/economy/economy.yaml:187]: I agree and will be aware of this unit issue for the next PR iteration.

**Comment by sebastianzwickl** [comment]
Excuse for inflationary commits. `f'` has challenged me ;-)

**Review by danielhuppmann — APPROVED** [review]
Thanks @sebastianzwickl for this nice improvement!

**Pull request merged**